Source/PINOperationGroup.m

```diff
@@ -167,6 +158,20 @@ - (void)waitUntilComplete
 {
   [self start];
   dispatch_group_wait(_group, DISPATCH_TIME_FOREVER);
+  [self runCompletionIfNeeded];
```

Comment:
Should we add documentation to the header that the completion may not run on the completion queue if called from waitUntilComplete? Otherwise should we use a semaphore to block this thread while dispatching out to the completion thread (I'm afraid of this idea).